### PR TITLE
Add a troubleshooting guide. close #643

### LINKED ISSUE #643: Add a troubleshooting guide
* [x] update Jackson to a current version, especially in Maven force version (toplevel pom needs explicit Jackson lib dependency)
* [x] AHC version check
* [x] log file flooding disable, async http client is chatty
* [x] Java version
* [x] os
* [ ] which data do we need before contact commercetools, provide 3 good examples

### PR DESCRIPTION
**Pull request by schleichardt**

### CODE REVIEW

**Comment by lauraluiz** [comment]
Should I review it?

**Comment by schleichardt** [comment]
Yes, I did not explicitly mentioned you since I wanted to wait until the Javadoc is deployed.

Now it is available: http://sphereio.github.io/sphere-jvm-sdk/javadoc/feature/643/troubleshooting-guide/io/sphere/sdk/meta/TroubleshootingGuide.html

@lauraluiz please review

;)

File: src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+
+  <h3 id="jackson-dependency">JSON Jackson Initialization Problems</h3>
+  <h4>Indications</h4>
+ {@code Could not initialize class <a commercetools related class>}
+
+ <h4>Possible solutions</h4>
+ It could be the case that you use a build tool like Maven which has other code in it depending on an older Jackson version.
+ The <a href="http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Transitive_Dependencies">Maven way of getting transitive dependencies</a> may cause that you
+ an outdated Jackson version and you need to explicitly pull it up like with:
```
Comment (lauraluiz) [inline src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java]: `that you *use* an outdated Jackson...` (use or any fitting synonym)

File: src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+<a href="https://github.com/AsyncHttpClient/async-http-client/blob/master/MIGRATION.md" target="_blank">A list of breaking changes from Ning Async HTTP Client 1.8 to 1.9.</a>
+
+
+ <h3 id=ahc-log-flooding>Ning Async Http Client is flooding the logs</h3>
+
+ See {@link io.sphere.sdk.utils.SphereInternalLogger}.
+
+ <h3 id=reuqired-data-to-reproduce>How to provide a reproduceable support question</h3>
+ If you get an exception, give use the stacktrace with data you want to share.
```
Comment (lauraluiz) [inline src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java]: `..., give use the stacktrace...` give or use?

File: src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+ [error]     at io.sphere.sdk.http.DefaultAsyncHttpClientAdapterImpl.lambda$wrap$5(DefaultAsyncHttpClientAdapterImpl.java:93)
+ [error]     at io.sphere.sdk.http.DefaultAsyncHttpClientAdapterImpl$$Lambda$75/644950868.run(Unknown Source)
+ [error]     at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
+ [error]     at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
+ [error]     at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
+ [error]     at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1689)
+ [error]     at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
+ }</pre>
+
+ <p>If you get not an exception, it may be of importance to know you JVM SDK version, Java version, date of occurrence and operating system.</p>
```
Comment (lauraluiz) [inline src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java]: `If you *do not get* an exception...`
`to know *your* JVM SDK version...`

File: src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+ Describe the behavior which you observed and which you would like to observe.
```
Comment (lauraluiz) [inline src/main/java/io/sphere/sdk/meta/TroubleshootingGuide.java]: `Describe the behavior you observed and the one you would like to obtain.`

**Pull request merged**